### ISSUE TITLE
Evaluate history

### ISSUE THREAD

**Issue by hklene** [alexpdraper/reading-list#52]
In the wake of #49 I expect the majority of links I add, won't be in the Browser-History. I'd like to suggest adding a marker, if I've seen some link before.

E.g. reading a news-site I may add articles from teasers on the front-page. Then I start reading some articles and get advertisements for even more interesting articles on the side of the text. Now I could click such a link and read an article, that is already in my reading list, but as I don't open it from there, I'm not aware.

1. Could you mark URLs, that have been visited before?
2. Can you remove single links, if exactly that link is (re-)loaded in some tab?
3. Could you offer a bulk-operation, to discard any links with read-state?

**Comment by alexpdraper** [comment]
Do you mean edit the page content to indicate that the link is on your list already? I think that’s outside the scope of this project. I think reading a page that is already on your list is a non-issue for most people. Automatically deleting an item from your list when it’s opened from somewhere outside the reading list seems dangerous as well. Most of our users probably don’t want us messing with their list in unexpected ways.

Adding a way to remove all “read” articles is something we could realistically do.

**Comment by hklene** [comment]
No, this is not about modifying page content, but about the presentation within the popup.

#59 added some mysterious "alternate read state" I couldn't make sense of until now. To distinguish, in this bug #52 I'll switch the wording to _visited_ to describe whether a link is in the history DB and would appear violet versus blue for _unvisited_ links.

So the list from the original suggestion would read:
1. Mark URLs in the popup violet / blue depending on whether they appear in the history DB.
2. Auto-removal of links from the reading list hooked into history DB, triggered, when new links are visited.
3. Manual cleanup of visited links, make history DB and reading-list disjoint.

No 1 could have been something like #59 did: auto-filter the long list by three buttons: All, visited, unvisited. But I can't say how this would integrate with the mysterious alternate read state.

No 2 would expand #50 so it's not necessary to go through the addon to still get a consistent result. There are always multiple ways to do things.

No 3 finally is a bulk operation for dealing with potentially hundreds of links at a time. You don't want to wipe the whole reading-list, but finish off, what has been visited. Maybe this could use some tuning, to make it a really sharp tool for working precisely. E.g. I imagine, further filtering the "visited" list by entering a domain in the search and only discarding whats been read on my favorite news site, leaving wikipedia-articles unharmed, though some of them might have a visited state.

**Comment by alexpdraper** [comment]
We’re not using people’s browser history for the read/unread state. We’re marking pages as read after someone visits them while they are on the reading list.

We could add a third button so we have: All, Unread, and Read.
